ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+    }
+
+    if (pMMArcLayer->MMNode.pF)
+        fclose_function(pMMArcLayer->MMNode.pF);
+
+    return 0;
+}
+
+static int MMCloseArcLayer(struct MiraMonVectLayerInfo *hMiraMonLayer)
+{
+    struct MiraMonArcLayer *pMMArcLayer;
+    struct MM_TH *pArcTopHeader;
+
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (hMiraMonLayer->bIsPolygon)
+    {
+        pMMArcLayer = &hMiraMonLayer->MMPolygon.MMArc;
+        pArcTopHeader = &hMiraMonLayer->MMPolygon.TopArcHeader;
+    }
+    else
+    {
+        pMMArcLayer = &hMiraMonLayer->MMArc;
+        pArcTopHeader = &hMiraMonLayer->TopHeader;
+    }
+
+    if (hMiraMonLayer->ReadOrWrite == MM_WRITTING_MODE)
+    {
+        hMiraMonLayer->nFinalElemCount = pArcTopHeader->nElemCount;
+        pArcTopHeader->bIs3d = hMiraMonLayer->bIsReal3d;
+
+        if (MMWriteHeader(pMMArcLayer->pF, pArcTopHeader))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    pMMArcLayer->pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+        hMiraMonLayer->OffsetCheck = hMiraMonLayer->nHeaderDiskSize;
+
+        // AH Section
+        if (MMWriteAHArcSection(hMiraMonLayer, hMiraMonLayer->OffsetCheck))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    pMMArcLayer->pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+
+        // AL Section
+        pMMArcLayer->FlushAL.SizeOfBlockToBeSaved = 0;
+        if (MMAppendBlockToBuffer(&pMMArcLayer->FlushAL))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    pMMArcLayer->pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+        if (MMMoveFromFileToFile(pMMArcLayer->pFAL, pMMArcLayer->pF,
+                                 &hMiraMonLayer->OffsetCheck))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    pMMArcLayer->pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+        if (pMMArcLayer->pFAL)
+            fclose_function(pMMArcLayer->pFAL);
+        if (*pMMArcLayer->pszALName != '\0')
+            remove_function(pMMArcLayer->pszALName);
+
+        // 3D Section
+        if (MMClose3DSectionLayer(
+                hMiraMonLayer, pArcTopHeader->nElemCount, pMMArcLayer->pF,
+                pMMArcLayer->pF3d, pMMArcLayer->psz3DLayerName,
+                &pMMArcLayer->pZSection, hMiraMonLayer->OffsetCheck))
+        {
+            sprintf(local_message, "Error writing to file %s",
+                    pMMArcLayer->pszLayerName);
+            MMCPLError(CE_Failure, CPLE_NoWriteAccess, local_message);
+            return 1;
+        }
+    }
+
+    if (pMMArcLayer->pF)
+        fclose_function(pMMArcLayer->pF);
```

Comment:
```suggestion
    {
        fclose_function(pMMArcLayer->pF);
        pMMArcLayer->pF = nullptr;
    }
```

Reply:
Actually as it is a common parttern, it might be relevant to have a fclose_and_nullify(FILE_TYPE **pFunc) that takes a pointer to a FILE_TYPE*, to make sure to nullify it

Reply:
Yes but it should be fclose_and_nullify(FILE_TYPE **pFunc). Shoulnd't it? (two *)

Reply:
void fclose_and_nullify(FILE_TYPE **pFunc)
{
    if(!pFunc)
        return;
    fclose_function(*pFunc);
    *pFunc=nullptr;
}

Reply:
> Yes but it should be fclose_and_nullify(FILE_TYPE **pFunc). Shoulnd't it? (two *)

yes, that's what I had written litteraly, but forgot to add proper markdown formatting, so it reads as a single star instead of the intended 2 stars

Reply:
Perfect. Used everywhere